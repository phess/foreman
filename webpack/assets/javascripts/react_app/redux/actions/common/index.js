import API from '../../../API';

export const ajaxRequestAction = ({
  dispatch,
  requestAction,
  successAction,
  failedAction,
  url,
  item = {},
}) => {
  dispatch({ type: requestAction, payload: item });
  return API.get(url, item.headers || {}, item.params || {})
    .then(({ data }) =>
      dispatch({ type: successAction, payload: { ...item, ...data } })
    )
    .catch(error => dispatch({ type: failedAction, payload: { error, item } }));
};
