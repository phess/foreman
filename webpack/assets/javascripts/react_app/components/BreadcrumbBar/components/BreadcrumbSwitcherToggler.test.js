import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';

import BreadcrumbSwitcherToggler from './BreadcrumbSwitcherToggler';

const fixtures = {
  'render a BreadcrumbSwitcherToggler': {
    onClick: jest.fn(),
  },
};

describe('BreadcrumbSwitcherToggler', () =>
  testComponentSnapshotsWithFixtures(BreadcrumbSwitcherToggler, fixtures));
