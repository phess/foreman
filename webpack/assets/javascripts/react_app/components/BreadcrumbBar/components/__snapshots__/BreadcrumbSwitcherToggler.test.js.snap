// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbSwitcherToggler render a BreadcrumbSwitcherToggler 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={false}
  onClick={[MockFunction]}
>
  <Icon
    name="exchange"
    type="fa"
  />
</Button>
`;
